File: Py/CountpairsSuminmatrices.py
def countpairs(mat1,mat2,n,x):
    i1=0
    j1=0
    i2=n-1
    j2=n-1
    count=0
    while(i1<n and i2>=0):
        ans=mat1[i1][j1]+mat2[i2][j2]
        if(ans==x):
            count+=1
            j1+=1
            j2-=1
        elif(ans<x):
            j1+=1
        else:
            j2-=1
        if(j1==n):
            j1=0
            i1+=1
        if(j2==-1):
            j2=n-1
            i2-=1
    return count

File: _Java/Count pairs Sum in matrices.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
//import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String input[] = read.readLine().split(" ");
            
            int n = Integer.parseInt(input[0]);
            int x = Integer.parseInt(input[1]);
            
            int mat1[][] = new int[n][n];
            
            for(int i = 0;i<n;i++){
                input = read.readLine().split(" ");
                for(int j = 0;j<n;j++){
                    mat1[i][j] = Integer.parseInt(input[j]);
                }
            }
            
            int mat2[][] = new int[n][n];
            
            for(int i = 0;i<n;i++){
                input = read.readLine().split(" ");
                for(int j = 0;j<n;j++){
                    mat2[i][j] = Integer.parseInt(input[j]);
                }
            }
            
        
            
            Solution ob = new Solution();
            System.out.println(ob.countPairs(mat1,mat2,n,x));
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution {
    int countPairs(int mat1[][], int mat2[][], int n, int x) {
        // code here
        int start = 0;
        int end=n*n-1;
        int no=0;
       while(start<n*n && end>=0){
            int a=mat1[start/n][start%n];
            int b=mat2[end/n][end%n];
            if(a+b==x){
                no++;
                start++;
                end--;
            }
            else if(a+b>x){
                end--;
            }
            else{
                start++;
            }
        }
        return no;
    }
}